src/traces/mesh3d/convert.js

```diff
@@ -71,19 +71,21 @@ proto.update = function(data) {
     var positions = zip3(
         toDataCoords(layout.xaxis, data.x, scene.dataScale[0], data.xcalendar),
         toDataCoords(layout.yaxis, data.y, scene.dataScale[1], data.ycalendar),
-        toDataCoords(layout.zaxis, data.z, scene.dataScale[2], data.zcalendar));
+        toDataCoords(layout.zaxis, data.z, scene.dataScale[2], data.zcalendar)
+    );
 
     var cells;
     if(data.i && data.j && data.k) {
-        cells = zip3(data.i, data.j, data.k);
-    }
-    else if(data.alphahull === 0) {
+        cells = zip3(
+            data.i.map(function(e) { return Math.round(e); }),
```

Comment:
Let's use a for-loop here instead of three `.map`s for speed.

Reply:
Good hint. Replaced those as well as other instances of using `map` in `mesh3d/convert.js` https://github.com/plotly/plotly.js/pull/3369/commits/83060bc00a5cc17b1144f486a01e02903b1b9524 .